models/quantity.rb

```diff
@@ -30,5 +30,13 @@ class Quantity < Base
     # validates_numericality_of :setup_value
 
     belongs_to :unit, class_name: 'Models::Unit', as_json: :id
+
+    def self.field(name, options = {})
+      # Active_hash did not record default values set to false
+      # because the condition to reach add_default_value is `if options[:default]`
+      super(name, options)
+      add_default_value(:empty, false)
+      add_default_value(:fill, false)
```

Comment:
it is a good idea to to override `field` for this purpose  :ok_hand: instead of create. I have 2 remarks:
1. don't you get `field` method is redefined warning by doing this? You might want to use `redefine_method` function to prevent these warnings.
2. what is the return value of `add_default_value()` function. if super is not the last call, the `field` function of quantity and others will behave differently (unless `add_default_value` returns the whole object). Can you verify this please?

Reply:
1. Where should i see these warnings ?

2. `field` method is not expected to return any value

Reply:
1. when you launch a test with an instance with a quantity for example? If there is no method redefinition warning, are you sure this method is called? Can you put a byebug and try?
2. your version of the field returns a value which is specifically `false`. I don't know what active_hash returns but my expectation is that we return the same thing and not a random false.

Reply:
1. There is now warning about the redefinition of `field`. But you are right this method is never called. But the behavior is correct without it. The default getter returns the default value `false`, and it doesn't returns the value within as_json as it is not part of the attributes
2. active_hash didn't return a particular value for method field. Its last step is to create a function : https://github.com/active-hash/active_hash/blob/1c95f992af3ec94c07e19846f042e12bd0b11dd1/lib/active_hash/base.rb#L224